### PR TITLE
Containerd alpha1 update

### PR DESCRIPTION
**Pull request by ijc**
From the interesting of the two commit messages:
```
    Synchronise vendoring with containerd v1.0.0-alpha1
    
    Some of these are arbitrary and just syncing for the sake of it, however the
    image- and runtime-spec are relevant. Interesting changes:
    
    - runtime spec:
      - LinuxRLimit is now POSIXRLimit.
      - Specs.Config is now a pointer.
      - LinuxResources.DisableOOMKiller moved to
        LinuxResources.LinuxMemory.DisableOOMKiller
    - image spec:
      - Platform.Features is removed (unused here).
```

### CODE REVIEW

**Comment by ijc** [comment]
Might be worth holding off on merging until I've done an end to end check with the corresponding linuxkit update.

**Comment by justincormack** [comment]
Rlimits and oom killer changes probably need moby/tool changes. Now runtime
spec is 1.0 would be good to use that tag anyway.

**Comment by ijc** [comment]
These are the moby/tool changes :-)

Wasn't sure re using the tagged spec or the version incorporated into containerd/runc.

**Comment by ijc** [comment]
runtime-spec v1.0.0 and image-spec v1.0.0-rc7 make no substantive changes (just the version and some comments) so I will bump.

**Comment by ijc** [comment]
Force pushed because I had previously forgotten a `git add vendor` after running `vndr`.

File: src/moby/config.go [hunk 1]
```diff
@@ -848,8 +848,15 @@ func ConfigInspectToOCI(yaml Image, inspect types.ImageInspect, idMap map[string
...
 			// Devices
-			DisableOOMKiller: assignBoolPtr(label.DisableOOMKiller, yaml.DisableOOMKiller),
-			// Memory
+			Memory: &specs.LinuxMemory{
+				// Limit
+				// Reservation
+				// Swap
+				// Kernel
+				// KernelTCP
+				// Swapiness
```
Comment (justincormack) [inline src/moby/config.go]: I think thats "swappiness"?

Comment (ijc) [inline src/moby/config.go]: It is, fixed.

File: vendor.conf [hunk 2]
```diff
@@ -5,22 +5,22 @@ github.com/docker/docker 6978a6e25a2e6063f280ec842bd0f3eae99426e1
...
-github.com/docker/go-units 651fc226e7441360384da338d0fd37f2440ffbe3
+github.com/docker/go-units v0.3.1
 github.com/docker/libtrust 9cbd2a1374f46905c68a4eb3694a130610adc62a
 github.com/docker/notary v0.4.2
 github.com/gogo/protobuf v0.4
 github.com/gorilla/mux 4c1c3952b7d9d0a061a3fa7b36fd373ba0398ebc
-github.com/opencontainers/go-digest a6d0ee40d4207ea02364bd3b9e8e77b9159ba1eb
-github.com/opencontainers/image-spec f03dbe35d449c54915d235f1a3cf8f585a24babe
+github.com/opencontainers/go-digest 21dfd564fd89c944783d00d069f33e3e7123c448
+github.com/opencontainers/image-spec v1.0.0-rc7
```
Comment (justincormack) [inline vendor.conf]: may as well Go to 1.0.0 here I think, probably no changes anyway, and we are barely using it.

Comment (ijc) [inline vendor.conf]: There is no v1.0.0 of the image spec tagged yet. Are you thinking of runtime spec?

**Comment by ijc** [comment]
Rebased again because my `origin/master` was woefully out of date (built, but couldn't be used to build the current `linuxkit.yml` due to missing `onshutdown`)

File: vendor.conf (hunk 2 above)
Comment (justincormack) [inline vendor.conf]: yes there is https://github.com/opencontainers/image-spec/releases

Comment (ijc) [inline vendor.conf]: Was confused by the sort order of `git tag -l` (non-rc is first)

**Comment by justincormack** [comment]
Looks ok, do you want to do a linuxkit PR too so they can be merged at roughly the same time?

**Comment by ijc** [comment]
linuxkit PR is about 90% complete. Should land in an hour or so.

**Comment by ijc** [comment]
@justincormack In fact it is ready now, all I need is the SHA of the merge of this PR to update with.

**Pull request merged**